### PR TITLE
Call onConnectionLost on all errors

### PR DESCRIPTION
**Pull request by insipx**
this calls the OnConnectionLost on all errors that might happen from grpc. This should let `messagekit` handle the error however it needs to

### CODE REVIEW

File: packages/bot-kit-pro/src/bot.test.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ describe("Bot", () => {
     })
     const bot = await Bot.create(config, dataSource)
     const otherClient = await Client.create(Wallet.createRandom(), {
-      apiClientFactory: GrpcApiClient.fromOptions,
+      apiClientFactory: GrpcApiClient.fromOptions as any,
```
Comment (insipx) [inline packages/bot-kit-pro/src/bot.test.ts:85]: not sure what best practice w.r.t node is but this is what was in messagekit so i just did that here

Comment (rygine) [reply packages/bot-kit-pro/src/bot.test.ts:85]: this is fine. bot kit pro is deprecated at this point.

**Review by rygine — APPROVED** [review]

File: .changeset/modern-laws-wait.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@xmtp/grpc-api-client": patch
+"@xmtp/bot-kit-pro": patch
```
Comment (rygine) [inline .changeset/modern-laws-wait.md:3]: you can remove this line so a new version of bot kit pro isn't published

File: packages/bot-kit-pro/src/bot.test.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ describe("Bot", () => {
     const wallet = Wallet.createRandom()
     keys = await Client.getKeys(wallet, {
       env: "dev",
-      apiClientFactory: GrpcApiClient.fromOptions,
+      apiClientFactory: GrpcApiClient.fromOptions as any,
```
Comment (rygine) [inline packages/bot-kit-pro/src/bot.test.ts:36]: are these `as any` assertions required?

**Pull request merged**